packages/docs/docs/guides/5-request-client/6-signature-provider.md

```diff
@@ -0,0 +1,224 @@
+This is fairly simple, you need to implement a class following this interface: (see on [github](https://github.com/RequestNetwork/requestNetwork/blob/master/packages/types/src/signature-provider-types.ts))
+
+```typescript
+export interface ISignatureProvider {
+  supportedMethods: Signature.METHOD[];
+  supportedIdentityTypes: Identity.TYPE[];
+
+  sign: (data: any, signer: Identity.IIdentity) => Promise<Signature.ISignedData>;
+}
+```
+
+## Example 1
+
+You have your own package to sign:
+
+```typescript
+class mySignaturePackage {
+  /**
+   * Sign data
+   *
+   * @param data the data to sign
+   * @param address the address to sign with
+   * @returns a promise resolving the signature
+   */
+  public async sign(data: any, address: string): Promise<string>;
+}
+```
+
+Your signature provider would look like:
+
+```typescript
+import { IdentityTypes, SignatureProviderTypes, SignatureTypes } from '@requestnetwork/types';
+import Utils from '@requestnetwork/utils';
+
+// Your package
+import mySignaturePackage from 'mySignaturePackage';
+
+/**
+ * Implementation of the signature provider for my wallet
+ */
+export default class MySignatureProvider implements SignatureProviderTypes.ISignatureProvider {
+  /** list of supported signing method */
+  public supportedMethods: SignatureTypes.METHOD[] = [SignatureTypes.METHOD.ECDSA];
+  /** list of supported identity types */
+  public supportedIdentityTypes: IdentityTypes.TYPE[] = [IdentityTypes.TYPE.ETHEREUM_ADDRESS];
+
+  /**
+   * Signs data
+   *
+   * @param string data the data to sign
+   * @returns IIdentity the identity to sign with if not given, the default signer will be used
+   *
+   * @returns string the signature
+   */
+  public async sign(
+    data: any,
+    signer: IdentityTypes.IIdentity,
+  ): Promise<SignatureTypes.ISignedData> {
+    if (!this.supportedIdentityTypes.includes(signer.type)) {
+      throw Error(`Identity type not supported ${signer.type}`);
+    }
+
+    // Hash the normalized data (e.g. avoid case sensitivity)
+    const hashData = Utils.crypto.normalizeKeccak256Hash(data).value;
+
+    // use your signature package
+    const signatureValue = mySignaturePackage.sign(hashData, signer.value);
+
+    return {
+      data,
+      signature: {
+        method: SignatureTypes.METHOD.ECDSA,
+        value: signatureValue,
+      },
+    };
+  }
+}
+```
+
+Now you can inject it into the request client:
+
+```typescript
+import MySignatureProvider from 'mySignatureProvider';
+
+const mySignatureProvider = new MySignatureProvider();
+
+// We can initialize the RequestNetwork class with the signature provider
+const requestNetwork = new RequestNetwork.RequestNetwork({
+  signatureProvider: mySignatureProvider,
+});
+```
+
+## Example 2
+
+You have your own package to sign:
```

Comment:
This line should have more precisions about the example. It makes it look like they are both the same example.